src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -273,7 +346,8 @@ public bool Equals(TypeRef other)
                 && AssemblyNameComparer.Equals(this.AssemblyName, other.AssemblyName)
                 && this.IsArray == other.IsArray
                 && this.GenericTypeParameterCount == other.GenericTypeParameterCount
-                && this.GenericTypeArguments.EqualsByValue(other.GenericTypeArguments);
+                && this.GenericTypeArguments.EqualsByValue(other.GenericTypeArguments)
+                && this.BaseTypes.EqualsByValue(other.BaseTypes);
             return result;
```

Comment:
Can you remind me why we need to check this? This strikes me as an expensive comparison, and as we've already confirmed by this point that the metadata tokens and generic type arguments are the same, why do we need to verify the base type hierarchy hasn't changed? If it's to distinguish between shallow and not-shallow instances, can't we just compare `this.BaseTypes.IsEmpty == other.BaseTypes.IsEmpty` instead as that would be super-quick?

Reply:
This was to distinguish shallow types from non-shallow types for the purposes of serialization. I can change this to just use `IsEmpty`.